sanic/router.py

```diff
@@ -163,7 +163,7 @@ def add_parameter(match):
             parameters.append(parameter)
 
             # Mark the whole route as unhashable if it has the hash key in it
-            if re.search(r'(^|[^^]){1}/', pattern):
+            if re.search(r'(^|[^^]){0}/', pattern):
```

Comment:
I think you might be confused by the `pattern` variable name? The first argument is a pattern, the second is the string which named `pattern`.